spec/integration/errors_spec.rb

```diff
@@ -8,18 +8,18 @@ def set_consider_all_requests_local(value)
     config = app.config
     env_config = app.env_config
 
-    allow(app).to receive(:env_config).with(no_args) do
```

Comment:
> Prefer toggling the config and restoring the vale after the spec example
> has run.

I can see _what_ this does, but it doesn't explain _why_ it's preferable to do this here.

Reply:
Until I split the commits up I thought this was necessary to get this working... I left this change as it was otherwise it would mean rewriting the later commits in the branch.